GameSessionsManager.h

```diff
@@ -0,0 +1,26 @@
+#ifndef TICTACTOE_GAMESESSIONSMANAGER_H
+#define TICTACTOE_GAMESESSIONSMANAGER_H
+
+#include <thread_pool.h>
+#include <GrpcDelegatingWriter.h>
+#include <GrpcDelegatingReader.h>
+#include <Player.h>
+#include "PlayerInfo.h"
+#include <GameSession.h>
+
+class GameSessionsManager {
+	ThreadPool game_sessions_pool;
+
+public:
+	void start_game_session(PlayerInfo player1, PlayerInfo player2) {
+		const Player& first_player = Player(player1.id, symbol::CROSS, new GrpcDelegatingWriter(player1.address), new GrpcDelegatingReader(player1.address));
+		const Player& second_player = Player(player2.id, symbol::CIRCLE, new GrpcDelegatingWriter(player2.address), new GrpcDelegatingReader(player2.address));
+
+		game_sessions_pool.submit_job([&]() {
+			GameSession session(first_player, second_player);
+			session.play();
+		});
```

Comment:
The lambda expression will outlive the variables it has been passed, so capturing by reference will produce dangling references.